.githubtemplate/workflows/deploy_staging_and_production.yml

```diff
@@ -0,0 +1,82 @@
+name: Deploy Staging
+
+on:
+  # Trigger the workflow on push action in develop branch
+  # So it will trigger when the PR of the feature branch was merged.
+  push:
+    branches:
+      - develop
+jobs:
+  app_deploy:
+    name: Deploy and upload to Firebase App Distribution
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Setup timezone
+        uses: zcong1993/setup-timezone@master
+        with:
+          timezone: Asia/Bangkok
+
+      - uses: actions/checkout@v2.3.2
+
+      - name: Load Gradle cached
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Run Detekt
+        run: ./gradlew detekt
+
+      - name: Archive Detekt report
+        uses: actions/upload-artifact@v2
+        with:
+          name: detekt-report
+          path: build/reports/detekt/
+
+      - name: Run Unit Tests & Jacoco
+        run: ./gradlew jacocoTestReport
+
+      # Archive Unit test & code coverage report
+      - name: Archive Code coverage report
+        uses: actions/upload-artifact@v2
+        with:
+          name: code-coverage-report
+          path: |
+            app/build/reports/jacoco/jacocoTestReport/
+            data/build/reports/jacoco/jacocoTestReport/
+
+      # Build Staging Debug
+      - name: Build Staging
+        run: ./gradlew assembleStagingDebug
+
+      # Upload artifact to Staging Firebase App Distribution
+      - name: Upload to Firebase
+        uses: wzieba/Firebase-Distribution-Github-Action@v1
+        with:
+          appId: ${{secrets.FIREBASE_APP_ID_STAGING}}
+          token: ${{secrets.FIREBASE_TOKEN}}
```

Comment:
maybe we can create a new `README.md` for showing how to setup these keys for the workflow to work, right now with just this and there is no setup values for these keys then for sure this workflow will fail. What do you think?